driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/AsyncTransportSettingsTest.java

```diff
@@ -51,20 +53,26 @@ void testAsyncTransportSettings() {
         verify(executorService, atLeastOnce()).execute(any());
     }
 
-    @Test
-    void testExternalExecutorNotShutDown() {
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void testExternalExecutorWasShutDown(final boolean tlsEnabled) {
         ExecutorService executorService = spy(Executors.newFixedThreadPool(5));
         AsyncTransportSettings asyncTransportSettings = TransportSettings.asyncBuilder()
                 .executorService(executorService)
                 .build();
         MongoClientSettings mongoClientSettings = getMongoClientSettingsBuilder()
-                .applyToSslSettings(builder -> builder.enabled(true))
+                .applyToSslSettings(builder -> builder.enabled(tlsEnabled))
                 .transportSettings(asyncTransportSettings)
                 .build();
 
         try (MongoClient ignored = new SyncMongoClient(MongoClients.create(mongoClientSettings))) {
             // ignored
         }
-        verify(executorService, never()).shutdown();
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
The spy doesn't pick up the ensuing shutdown otherwise.

Reply:
https://jira.mongodb.org/browse/JAVA-5703

Reply:
1. As far as I can see, we don't need `Mockito.spy` in this test, because there are `ExecutorService.isShutDown`/`isTerminated` methods.
2. The reason an executor is not shutdown synchronously in this test, is that `com.mongodb.internal.connection.tlschannel.async.AsynchronousTlsChannelGroup.shutdown` only "_Starts the shutdown process_", as per the documentation. It wakes up the selector, which allows the `while` loop in the `AsynchronousTlsChannelGroup.loop` method to continue and call `executor.shutdown`, but that happens in a thread different from the one executing `AsynchronousTlsChannelGroup.shutdown`.
3. Let's use `assertTrue(executorService.awaitTermination(100, TimeUnit.MILLISECONDS))`.